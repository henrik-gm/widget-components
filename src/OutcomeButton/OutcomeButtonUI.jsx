import React from 'react';

/**
 * Renders the UI of an outcome button, no
 * special handling logic.
 */
const OutcomeButtonUI = ({ label, odds, suspended, selected, onClick }) => {
   let buttonCssClasses = 'l-flex-1 kw-link KambiWidget-outcome';
   if (suspended) {
      buttonCssClasses += ' KambiWidget-outcome--suspended';
   } else if (selected) {
      buttonCssClasses += ' KambiWidget-outcome--selected';
   }
   if (label == null) {
      return (
         <button
            type="button"
            role="button"
            className={buttonCssClasses}
            disabled={suspended}
            onClick={onClick}
         >
            <div className="l-flexbox l-pack-center">
               <div className="KambiWidget-outcome__odds-wrapper">
                  <span className="KambiWidget-outcome__odds">
                     {odds}
                  </span>
               </div>
            </div>
         </button>
      );

   } else {

      return (
         <button
            type="button"
            role="button"
            className={buttonCssClasses}
            disabled={suspended}
            onClick={onClick}
         >
            <div className="KambiWidget-outcome__flexwrap">
               <div className="KambiWidget-outcome__label-wrapper">
                  <span className="KambiWidget-outcome__label">
                     {label}
                  </span>
                  <span className="KambiWidget-outcome__line" />
               </div>
               <div className="KambiWidget-outcome__odds-wrapper">
                  <span className="KambiWidget-outcome__odds">
                     {odds}
                  </span>
               </div>
            </div>
         </button>
      );
   }
}

OutcomeButtonUI.propTypes = {
   /**
    * If null changes the layout of the button to not show a label
    * If string uses that as the label
    */
   label: React.PropTypes.string,

   /**
    * Odds to show
    */
   odds: React.PropTypes.string.isRequired,

   /**
    * If the button is enabled or not (greyed out)
    */
   suspended: React.PropTypes.bool,

   /**
    * If the button is selected or not
    */
   selected: React.PropTypes.bool.isRequired,

   /**
    * Callback for when the button is clicked
    */
   onClick: React.PropTypes.instanceOf(Function)
};

OutcomeButtonUI.defaultProps = {
   label: null,
   suspended: false
};

export default OutcomeButtonUI;
