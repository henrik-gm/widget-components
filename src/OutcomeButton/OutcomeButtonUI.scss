.label {
   justify-content: center;
}

.general {
   cursor: pointer;
   flex: 1;
}

.outline {
   background: transparent;
   border: 1.2px solid rgba(#fff, .8);
   box-sizing: border-box;
   padding: -1.2px;

   &:hover {
      background: rgba(#fff, .1);
      border: 1.2px solid #fff;
      padding: -1.2px;
   }
}

.outline--selected {
   border: 1.2px solid transparent;
}

.outline__odds {
   color: #fff;
}

.outline__label-wrapper {
   color: #fff;
   opacity: .85;
}