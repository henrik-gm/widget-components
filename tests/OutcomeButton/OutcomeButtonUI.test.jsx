/* eslint-env jest */
import React, { Children } from 'react';
import OutcomeButtonUI from '../../src/OutcomeButton/OutcomeButtonUI';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

let renderer;

describe('OutcomeButtonUI DOM rendering', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly when neither selected nor suspended', () => {
      expect(renderer.render(
         <OutcomeButtonUI odds='2.0' selected={false} suspended={false} />
      )).toMatchSnapshot();
   });

   it('renders correctly when selected', () => {
      expect(renderer.render(
         <OutcomeButtonUI odds='2.0' selected={true} />
      )).toMatchSnapshot();
   });

   it('renders correctly when selected', () => {
      expect(renderer.render(
         <OutcomeButtonUI odds='2.0' selected={false} suspended={true} />
      )).toMatchSnapshot();
   });

   it('throws error when both odds and label are not provided', () => {
      expect(() => {
         renderer.render(
            <OutcomeButtonUI selected={true}/>
         );
      }).toThrowErrorMatchingSnapshot();
   });

   it('renders correctly with label', () => {
      expect(renderer.render(
         <OutcomeButtonUI label='Test label' selected={false} />
      )).toMatchSnapshot();
   });

   it('renders correctly with odds and label', () => {
      expect(renderer.render(
         <OutcomeButtonUI odds='5.0' label='Test label' selected={false} />
      )).toMatchSnapshot();
   });

});

describe('OutcomeButtonUI behaviour', () => {

   it('handles clicks correctly', () => {
      const onClickMock = jest.fn();

      const wrapper = shallow(
         <OutcomeButtonUI odds='1.2' selected={true} onClick={onClickMock} />
      );

      expect(onClickMock).not.toHaveBeenCalled();

      wrapper.find('button').simulate('click');
      expect(onClickMock).toHaveBeenCalledTimes(1);
   });

});
