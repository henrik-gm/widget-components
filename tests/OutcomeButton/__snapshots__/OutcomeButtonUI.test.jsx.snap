// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutcomeButtonUI DOM rendering renders correctly when neither selected nor suspended 1`] = `
<button
  className="l-flex-1 kw-link KambiWidget-outcome general"
  disabled={false}
  onClick={undefined}
  role="button"
  type="button"
>
  <div
    className="KambiWidget-outcome__odds-wrapper"
  >
    <span
      className="KambiWidget-outcome__odds"
    >
      2.0
    </span>
  </div>
</button>
`;

exports[`OutcomeButtonUI DOM rendering renders correctly when selected 1`] = `
<button
  className="l-flex-1 kw-link KambiWidget-outcome general KambiWidget-outcome--selected"
  disabled={false}
  onClick={undefined}
  role="button"
  type="button"
>
  <div
    className="KambiWidget-outcome__odds-wrapper"
  >
    <span
      className="KambiWidget-outcome__odds"
    >
      2.0
    </span>
  </div>
</button>
`;

exports[`OutcomeButtonUI DOM rendering renders correctly when selected 2`] = `
<button
  className="l-flex-1 kw-link KambiWidget-outcome general KambiWidget-outcome--suspended"
  disabled={true}
  onClick={undefined}
  role="button"
  type="button"
>
  <div
    className="KambiWidget-outcome__odds-wrapper"
  >
    <span
      className="KambiWidget-outcome__odds"
    >
      2.0
    </span>
  </div>
</button>
`;

exports[`OutcomeButtonUI DOM rendering renders correctly with label 1`] = `
<button
  className="l-flex-1 kw-link KambiWidget-outcome general"
  disabled={false}
  onClick={undefined}
  role="button"
  type="button"
>
  <div
    className="KambiWidget-outcome__label-wrapper label"
  >
    <span
      className="KambiWidget-outcome__label"
    >
      Test label
    </span>
  </div>
</button>
`;

exports[`OutcomeButtonUI DOM rendering renders correctly with odds and label 1`] = `
<button
  className="l-flex-1 kw-link KambiWidget-outcome general"
  disabled={false}
  onClick={undefined}
  role="button"
  type="button"
>
  <div
    className="KambiWidget-outcome__flexwrap"
  >
    <div
      className="KambiWidget-outcome__label-wrapper"
    >
      <span
        className="KambiWidget-outcome__label"
      >
        Test label
      </span>
      <span
        className="KambiWidget-outcome__line"
      />
    </div>
    <div
      className="KambiWidget-outcome__odds-wrapper"
    >
      <span
        className="KambiWidget-outcome__odds"
      >
        5.0
      </span>
    </div>
  </div>
</button>
`;

exports[`OutcomeButtonUI DOM rendering throws error when both odds and label are not provided 1`] = `"Both odds and label cannot be set to null"`;
